packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx

```diff
@@ -148,6 +148,8 @@ export const ArrayInput = (props: ArrayInputProps) => {
             </Labeled>
         );
     }
+    const renderHelperText =
+        helperText !== false || ((isDirty || isSubmitted) && !!error);
```

Comment:
This condition is different from the one before.

If `helperText = undefined` (the default):
- Before we would have rendered `null`
- Now we render `<FormHelperText ... />`

This creates an unnecessary additional gap.
You should keep the previous condition IMO.

Reply:
It was wrong before. The additionnal gap isn't unnecessary as it prevents the UI from jumping when error messages are displayed